emcc.py

```diff
@@ -62,16 +62,17 @@
 CXX_ENDINGS = ('.cpp', '.cxx', '.cc', '.c++', '.CPP', '.CXX', '.C', '.CC', '.C++', '.ii')
 OBJC_ENDINGS = ('.m', '.mi')
 OBJCXX_ENDINGS = ('.mm', '.mii')
+ASSEMBLY_CPP_ENDINGS = ('.S',)
```

Comment:
What is the difference between `ASSEMBLY_CPP_ENDINGS` and `ASSEMBLY_ENDINGS`? Should we allow `.asm` as well?

Reply:
`.S` files run through the C pre-processor, so things `-D` make sense.  `.s` files are not pre-possessed and clang will actually warn if you have any `-D` flags.